aspect/src/main/java/tech/ydb/yoj/aspect/tx/YojTransactionAspect.java

```diff
@@ -48,12 +48,27 @@ private Object doInTransaction(ProceedingJoinPoint pjp, YojTransactional transac
             if (transactional.maxRetries() != YojTransactional.UNDEFINED) {
                 localTx = tx.withMaxRetries(transactional.maxRetries());
             }
-            return transactional.readOnly() ? localTx.readOnly().run(() -> safeCall(pjp)) : localTx.tx(() -> safeCall(pjp));
+
+            validateIsolationLevel(transactional);
+
+            if (transactional.readOnly()) {
+                return localTx.readOnly()
```

Comment:
style: let's move `.readOnly()` to the next line

Reply:
Done